go.mod

```diff
@@ -1,12 +1,12 @@
 module github.com/hashicorp/terraform-plugin-mux
 
-go 1.20
+go 1.21
 
 require (
 	github.com/google/go-cmp v0.6.0
-	github.com/hashicorp/terraform-plugin-go v0.21.0
+	github.com/hashicorp/terraform-plugin-go v0.22.0
```

Comment:
Running `go get github.com/hashicorp/terraform-plugin-go` results in the addition of the following entry to `go.mod`:

```
toolchain go1.21.6
```

Resulting in a [CI failure](https://github.com/hashicorp/terraform-plugin-mux/actions/runs/8017456549/job/21901361119#step:4:7):

```
go: errors parsing go.mod:
/home/runner/work/terraform-plugin-mux/terraform-plugin-mux/go.mod:[5](https://github.com/hashicorp/terraform-plugin-mux/actions/runs/8017456549/job/21901361119#step:4:6): unknown directive: toolchain
```

Given that our CI runs both Go `1.20` and `1.21`, I removed this directive, but perhaps this should be handled differently?